test/message_processor_test.rb

```diff
@@ -0,0 +1,84 @@
+require "./test/test_helper"
+require "minitest/autorun"
+require "minitest/pride"
+require "./lib/message_processor"
+
+class MessageProcessorTest < Minitest::Test
+
+  def setup
+    @message_processor = MessageProcessor.new("hello world", "02715", "040895")
+    @message_processor2 = MessageProcessor.new("hello1, world!@%^&*", "02715", "040895")
```

Comment:
I will do more fringe testing in at a higher level, probably enigma class. Will check things such as negative numbers on key and date, also need to find a way to validate date format